pkg/controller/gcppubsub/provider.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gcppubsub
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/knative/eventing-sources/pkg/apis/sources/v1alpha1"
+	"github.com/knative/eventing-sources/pkg/controller/sdk"
+	"k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+const (
+	// controllerAgentName is the string used by this controller to identify
+	// itself when creating events.
+	controllerAgentName = "gcp-pubsub-source-controller"
+
+	// gcpPubSubEnabledEnvVar is used to determine if the GCP PubSub Source's controller should run.
+	// It will only run if the environment variable is defined and has the value 'true'.
+	gcpPubSubEnabledEnvVar = "ENABLE_GCPPUBSUB_SOURCE"
+
+	raImageEnvVar          = "GCPPUBSUB_RA_IMAGE"
```

Comment:
can you comment where these come from? it was not clear that this was not a user provided value

Reply:
:+1: to comments explaining these env vars.

Reply:
Done.